### PR TITLE
166187854 send verification email upon successful registraion

### PR DESCRIPTION
**Pull request by alexxsanya**
#### What does this PR do?
- send verification email upon successful registration

#### Description of Task to be completed?

- add email configuration
- add a method to generate verification link
- add a method to send a verification link
- add functionality to verify and activate a user
- add tests for the send verification functions
- improve test coverage
- add the staticfiles folder to fix staticfiles warning

#### How should this be manually tested?
- fetch this repo locally in terminal with `$ git fetch origin ft-verifcation-email-166187854`
- do `$ pip install -r requirements.txt` to install the added package dependency
- add the lines below to your env variable files and set them according the slack message I shared.
```
export SECURITY_PASSWORD_SALT='any_security_password_salt'
export EMAIL_HOST='your_email_host_used'
export EMAIL_USE_TLS='boolean (True/False)'
export EMAIL_PORT='your_email_port
export EMAIL_HOST_USER='you_email_address'
export HOST_PASSWORD='your_email_password'
```
- Now, run the application and then test `registration api`. upon successful registration, you will be asked to check your email for verification link as you may see below
Screenshot 2019-06-08 at 23 50 15

- The you should be able to see the email with the link.
Screenshot 2019-06-08 at 23 52 08

- After clicking on the link, you receive a message to confirm to you that the account has been activated.
Screenshot 2019-06-08 at 23 53 48

#### What are the relevant pivotal tracker stories?

- Story ID - [166187854](https://www.pivotaltracker.com/story/show/166187854)

#### DEV NOTE
- To have the tests to run ensure to edit your **SECRET_KEY** & **SECURITY_PASSWORD_SALT** to the one I will share in slack
- **Kindly remember to test all of the current APIs both locally and using the review app [link](https://ah-backend-realers-stagi-pr-14.herokuapp.com/)**

### CODE REVIEW

File: authors/settings.py [hunk 1]
```diff
@@ -179,3 +179,11 @@
 #Update Database Configuration
 prod_db  =  dj_database_url.config(conn_max_age=500)
 DATABASES['default'].update(prod_db)
+
+#EMAIL CONFIGURATIONS
+EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
+EMAIL_HOST = 'smtp.gmail.com'
+EMAIL_USE_TLS = True
+EMAIL_PORT = 587
+EMAIL_HOST_USER = 'thy.realers@gmail.com'
```
Comment (HabibSentongo) [inline authors/settings.py]: I suggest you make the `EMAIL_HOST_USER` and `EMAIL_HOST` be picked from the environment also. For easy switching when the need arises.

**Review by HabibSentongo — APPROVED** [review]
I have followed your directions and tested this manually and confirmed that it is functioning well. Good work.

File: authors/settings.py (hunk 1 above)
Comment (alexxsanya) [reply authors/settings.py]: @HabibSentongo this is a nice suggestion. Let me implement it.

Comment (alexxsanya) [reply authors/settings.py]: @HabibSentongo  I have worked on this.

**Review by neelxie — CHANGES_REQUESTED** [review]
I have successfully been registered.
Screenshot 2019-06-10 at 09 38 16
I receive the email and get my account verified
Screenshot 2019-06-10 at 09 39 36
This is the resulting error when i login in.
Screenshot 2019-06-10 at 09 42 05

**Comment by Emmanuelaaron** [comment]
Greatwork @alexxsanya following the steps you've given, I'm able to activate the account.

Screenshot 2019-06-10 at 10 05 11
Though, I really do not know why I get an error while using the local host. What could I not be doing right?
Screenshot 2019-06-10 at 10 06 13

**Comment by alexxsanya** [comment]
> I have successfully been registered.
> Screenshot 2019-06-10 at 09 38 16
> I receive the email and get my account verified
> Screenshot 2019-06-10 at 09 39 36
> This is the resulting error when i login in.
> Screenshot 2019-06-10 at 09 42 05

@neelxie I notice that the path you are using for the login endpoint is incorrect. kindly add a slash at the end at try again.

**Review by neelxie — APPROVED** [review]
I do realise that I had used a wrong url for login after rectifying it, it has worked. Thank you and good work.
Screenshot 2019-06-10 at 09 48 39
When i check out my user
Screenshot 2019-06-10 at 09 50 41

**Review by Emmanuelaaron — APPROVED** [review]
@alexxsanya  Great job

**Pull request merged**